ql/src/java/org/apache/hadoop/hive/ql/metadata/SessionHiveMetaStoreClient.java

```diff
@@ -1559,36 +1562,47 @@ public boolean dropPartition(String catName, String dbName, String tableName, St
   }
 
   @Override
-  public List<Partition> dropPartitions(String catName, String dbName, String tblName,
-      List<Pair<Integer, byte[]>> partExprs, PartitionDropOptions options, EnvironmentContext context)
+  public List<Partition> dropPartitions(TableName tableName,
```

Comment:
what happens if someone invokes the legacy signature? it's gonna call the delegate drectly?
````
  public boolean dropPartition(String catName, String db_name, String tbl_name, List<String> part_vals,
      PartitionDropOptions options) throws NoSuchObjectException, MetaException, TException {
    return delegate.dropPartition(catName, db_name, tbl_name, part_vals, options);
  }
````

Reply:
All the `dropPartitions` APIs will call the new `dropPartitions(TableName...)` api, ideally it will not miss the tempTable logic for `SessionHiveMetaStoreClient`.

Reply:
but in this case shouldn't we disable the old API as it could lead to incorrect behaviour?

Reply:
👍 now is good